src/plots/cartesian/set_convert.js

```diff
@@ -371,49 +381,99 @@ module.exports = function setConvert(ax, fullLayout) {
                 }
             }
 
-            // [ [cnt, {$cat: index}], for 1,2 ]
-            var seen = [[0, {}], [0, {}]];
-            // [ [arrayIn[0][i], arrayIn[1][i]], for i .. N ]
-            var list = [];
-
+            var axLabels = [];
+            var fullObjectList = [];
+            var cols = [];
+            // Don't think that the trace should be drawn at all if the lengths don't match. Removing the arrays length check. It is better to fail loudly than silently.
```

Comment:
For better or worse that's not the approach plotly.js takes. Partly because it gets used in manual data entry applications and other cases where a plot gets drawn before the data is complete, we almost always do our best to NOT fail, and just interpret as much of the data as we can and draw that.